storage/remote/write_handler.go

```diff
@@ -496,7 +496,7 @@ func NewOTLPWriteHandler(logger *slog.Logger, reg prometheus.Registerer, appenda
 	}
 
 	pl, err := otel.Use(reg,
-		otel.Processor(deltatocumulative.NewFactory(), nil), // TODO: how to pass config?
+		otel.Maybe(delta, otel.Processor(deltatocumulative.NewFactory(), nil)),
 		otel.Sink(ex),
 	)
```

Comment:
Do we really need this Maybe abstraction? Couldn't we just conditionally add things to the array?

```go
steps := make([]steps, 0, 2)
if delta {
  steps = append(steps, otel.Processor(deltatocumulative.NewFactory(), nil)
}
steps = append(steps, otel.Sink(ex))

pl, err := otel.Use(reg, ...steps)
```

Reply:
yes we could.

I preferred this, because it allows to cleanly read the pipeline top to bottom, instead of going through (diffrently indented) if and append calls.

I simplified the abstraction (returns either the step or nil now, that's it). Is this acceptable for you?